build_system/src/test.rs

```diff
@@ -13,50 +13,50 @@ use std::path::{Path, PathBuf};
 use std::str::FromStr;
 
 type Env = HashMap<String, String>;
-type Runner = &'static dyn Fn(&Env, &TestArg) -> Result<(), String>;
+type Runner = fn(&Env, &TestArg) -> Result<(), String>;
 type Runners = HashMap<&'static str, (&'static str, Runner)>;
 
 fn get_runners() -> Runners {
     let mut runners = HashMap::new();
 
     runners.insert(
         "--test-rustc",
-        ("Run all rustc tests", &test_rustc as Runner),
+        ("Run all rustc tests", test_rustc as Runner),
```

Comment:
The cast `as Runner` is probably not necessary anymore.

Reply:
No, this one is still needed.

Reply:
```
rror[E0308]: mismatched types
  --> src/test.rs:28:40
   |
28 |         ("Run successful rustc tests", test_successful_rustc),
   |                                        ^^^^^^^^^^^^^^^^^^^^^ expected fn item, found a different fn item
   |
   = note: expected fn item `for<'a, 'b> fn(&'a HashMap<String, String>, &'b TestArg) -> Result<_, _> {test_rustc}`
              found fn item `for<'a, 'b> fn(&'a HashMap<String, String>, &'b TestArg) -> Result<_, _> {test_successful_rustc}`

error[E0308]: mismatched types
  --> src/test.rs:32:37
   |
32 |         ("Run failing rustc tests", test_failing_rustc),
   |                                     ^^^^^^^^^^^^^^^^^^ expected fn item, found a different fn item
   |
   = note: expected fn item `for<'a, 'b> fn(&'a HashMap<String, String>, &'b TestArg) -> Result<_, _> {test_rustc}`
              found fn item `for<'a, 'b> fn(&'a HashMap<String, String>, &'b TestArg) -> Result<_, _> {test_failing_rustc}`
...
```